src/Microsoft.AspNet.Http/Authentication/DefaultAuthenticationManager.cs

```diff
@@ -62,75 +61,65 @@ public override IEnumerable<AuthenticationDescription> GetAuthenticationSchemes(
 
             if (!context.Accepted)
             {
-                throw new InvalidOperationException($"The following authentication scheme was not accepted: {context.AuthenticationScheme}");
+                throw new InvalidOperationException($"No authentication handler is configured to authenticate for the scheme: {context.AuthenticationScheme}");
             }
         }
 
-        public override async Task ChallengeAsync(string authenticationScheme, AuthenticationProperties properties, ChallengeBehavior behavior)
+        public override async Task ChallengeAsync(ChallengeContext context)
         {
-            if (authenticationScheme == null)
+            if (context == null)
             {
-                throw new ArgumentNullException(nameof(authenticationScheme));
+                throw new ArgumentNullException(nameof(context));
             }
 
             var handler = HttpAuthenticationFeature.Handler;
 
-            var challengeContext = new ChallengeContext(authenticationScheme, properties?.Items, behavior);
             if (handler != null)
             {
-                await handler.ChallengeAsync(challengeContext);
+                await handler.ChallengeAsync(context);
             }
 
-            if (!challengeContext.Accepted)
+            if (!context.Accepted)
             {
-                throw new InvalidOperationException($"The following authentication scheme was not accepted: {authenticationScheme}");
+                throw new InvalidOperationException($"No authentication handler is configured to handle a challenge for the scheme: {context.AuthenticationScheme}");
```

Comment:
I guess we can special case that and replace empty or null with something like automatic authentication